src/common/Database/DatabaseWorkerPool.h

```diff
@@ -469,6 +476,102 @@ class DatabaseWorkerPool
             return _connectionInfo.database.c_str();
         }
 
+        bool ExecuteFile(char const* file)
+        {
+            T* t = GetFreeConnection();
+
+            auto handle = t->GetHandle();
+            if (!handle)
+                return false;
+
+            if (mysql_set_server_option(handle, MYSQL_OPTION_MULTI_STATEMENTS_ON))
+            {
+                sLog->outErrorDb("Cannot turn multi-statements on: %s", mysql_error(handle));
+                return false;
+            }
+
+            mysql_autocommit(handle, 0);
+            if (mysql_real_query(handle, "START TRANSACTION", sizeof("START TRANSACTION") - 1))
+            {
+                sLog->outErrorDb("Couldn't start transaction for db update file %s", file);
+                return false;
+            }
+
+            bool in_trasaction = true;
+            bool success = false;
+
+            if (FILE* fp = ACE_OS::fopen(file, "rb"))
```

Comment:
I think we should avoid adding more code that depends on ACE (ideally we should one day get rid of ACE since it can just be replaced with modern C++). I'm sure there are plenty of ways to read files with modern C++ without using ACE or any library. The same apply to all other parts where ACE is used in this PR.

Reply:
indeed